### PR TITLE
fix(secrets):  import git

### PR DESCRIPTION
**Pull request by lirshindalman**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

Failed to initialize: Bad git executable

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Eliran-Turgeman** [comment]
I think we should set env var: GIT_PYTHON_REFRESH=quiet
https://github.com/gitpython-developers/GitPython/issues/816

File: checkov/secrets/scan_git_history.py [hunk 1]
```diff
@@ -1,16 +1,26 @@
...
 
 if TYPE_CHECKING:
     from detect_secrets import SecretsCollection
 
+os.environ["GIT_PYTHON_REFRESH"] = "quiet"
+try:
+    import git
+    git_import_error = None
+except ImportError as e:
+    git_import_error = e
```
Comment (marynaKK) [inline checkov/secrets/scan_git_history.py:16]: can we add here a log for the import error also, not only in the function? :)

**Review by marynaKK — APPROVED** [review]
🍒

File: checkov/secrets/scan_git_history.py (hunk 1 above)
Comment (lirshindalman) [reply checkov/secrets/scan_git_history.py:16]: I don't want to add the same log twice. if we don't use the function, the warning is irrelevant.

**Review by Eliran-Turgeman — APPROVED** [review]

**Pull request merged**